src/sentence/Sentence.spec.ts

```diff
@@ -223,6 +224,12 @@ describe('Sentence.js', () => {
         const { value } = new Sentence(helloWorldTemplate, helloWorldVocab);
         expect(value).toEqual('Hello, world.');
       });
+
+      it('should be able to use special characters', () => {
```

Comment:
Might consider having this test the `findPlaceholders()` function directly

Reply:
Agreed. To that end I've pulled that function as well as several others out of the Sentence class and put it into dedicated utils instead.